packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -331,14 +343,23 @@ class Environment {
     Directory flutterRootDir,
     Directory buildDir,
     Map<String, String> defines = const <String, String>{},
+    FileSystem fileSystem,
+    Logger logger,
+    @required Artifacts artifacts,
+    @required ProcessManager processManager,
   }) {
+    fileSystem ??= MemoryFileSystem.test();
     return Environment(
       projectDir: projectDir ?? testDirectory,
       outputDir: outputDir ?? testDirectory,
       cacheDir: cacheDir ?? testDirectory,
       flutterRootDir: flutterRootDir ?? testDirectory,
       buildDir: buildDir,
       defines: defines,
+      fileSystem: fileSystem,
+      logger: logger ?? BufferLogger.test(),
```

Comment:
This feels a little error-prone to me. Forgetting to pass a `BufferLogger` would lead to mysteriously missing log messages. The same sort of thing could be said for defaulting to `MemoryFileSystem.test()` above, too.

Reply:
Done